packages/core/src/modules/dids/domain/DidDocument.ts

```diff
@@ -201,11 +227,23 @@ export class DidDocument {
   }
 
   public get agreementKeys(): Array<VerificationMethod> {
-    return (
-      this.keyAgreement?.map((keyAgreement) => {
-        return typeof keyAgreement === 'string' ? this.dereferenceVerificationMethod(keyAgreement) : keyAgreement
-      }) ?? []
-    )
+    return this.dereferenceVerificationMethods(['keyAgreement'])
+  }
+
+  public get authentications(): Array<VerificationMethod> {
+    return this.dereferenceVerificationMethods(['authentication'])
+  }
+
+  public get assertionMethods(): Array<VerificationMethod> {
+    return this.dereferenceVerificationMethods(['assertionMethod'])
+  }
+
+  public get capabilityInvocations(): Array<VerificationMethod> {
+    return this.dereferenceVerificationMethods(['capabilityInvocation'])
+  }
+
+  public get capabilityDelegations(): Array<VerificationMethod> {
+    return this.dereferenceVerificationMethods(['capabilityDelegation'])
```

Comment:
This is a bit confusing to me. adding the `s` to make it derefenced is not really clear. Mabe we can just keep it at the single `dereferenceVerificationMethods` method and use it like that? 
Otherwise I think we should call it `getDereferencedAuthentication`, `getDerferencedAssertionMethod`, etc..